pdns/dnsdistdist/tcpiohandler.cc

```diff
@@ -431,6 +432,25 @@ class OpenSSLTLSIOCtx: public TLSCtx
         ERR_print_errors_fp(stderr);
         throw std::runtime_error("Error loading key from " + pair.second + " for the TLS context on " + fe.d_addr.toStringWithPort());
       }
+      if (SSL_CTX_check_private_key(d_tlsCtx.get()) != 1) {
+        ERR_print_errors_fp(stderr);
+        throw std::runtime_error("Key from '" + pair.second + "' does not match the certificate from '" + pair.first + "' for the TLS context on " + fe.d_addr.toStringWithPort());
+      }
+
+      /* store the type of the new key, we might need it later to select the right OCSP stapling response */
+      keyTypes.push_back(libssl_get_last_key_type(d_tlsCtx));
```

Comment:
Maybe we should check `libssl_get_last_key_type()` returned something positive ?